common/src/main/java/com/microsoft/identity/common/nativeauth/internal/controllers/NativeAuthMsalController.kt

```diff
@@ -704,8 +703,17 @@ class NativeAuthMsalController : BaseNativeAuthController() {
         // other tenants. Those tokens will be 'sparse', meaning that their AT/RT will not be loaded
         val fullCacheRecord = cacheRecords[0]
 
-        if (accessTokenIsNull(fullCacheRecord)) {
-            throw ServiceException(ErrorStrings.NATIVE_AUTH_NO_ACCESS_TOKEN_FOUND, "No access token found during refresh - user must be signed out.", null)
+        if (!accessTokenIsNull(fullCacheRecord) && silentTokenCommandParameters.scopes == null) {
```

Comment:
There may be an unhandled case here - if an access token is not returned, and there are scopes present in the `silentTokenCommandParameters`, we would no longer throw a service exception. Is this the expected behaviour?

Reply:
If we cannot find an access token for the given scopes then we will request a new access token from the server.